base_layer/core/src/base_node/state_machine_service/states/block_sync.rs

```diff
@@ -335,6 +336,11 @@ async fn synchronize_blocks<B: BlockchainBackend + 'static>(
                 request_and_add_blocks(shared, sync_peers, block_nums).await?;
                 sync_height += block_nums_count;
             }
+            let metadata = async_db::get_chain_metadata(shared.db()).await?;
+            let last_block = async_db::fetch_block(shared.db(), metadata.height_of_longest_chain()).await?;
+            shared
+                .local_node_interface
+                .publish_block_event(BlockEvent::BlockSyncComplete(Arc::new(last_block.block)));
```

Comment:
This is a problem for clearing out the mempool. 
We need to send through all the blocks we added, not just the last one. 
If we downloaded more than 1 block, only the last one will be cleared out. 

From the workings of the chain, a chain_reorg happend, that logic handles more than one block. We need to follow that logic here.

Reply:
Ye I thought so. This mempool logic really kills the node during block sync - perhaps we can reconcile the mempool after sync.

Reply:
I think reconciling the mempool afterwards makes more sense. Events are not guaranteed delivery. A more robust approach would be to only reconcile when you know the best chain data is in the correct state